Sources/SwiftKafka/Configuration/KafkaConfiguration.swift

```diff
@@ -0,0 +1,274 @@
+
+        public init(
+            addressTTL: Int = 1000,
+            addressFamily: KafkaConfiguration.IPAddressFamily = .any
+        ) {
+            self.addressTTL = addressTTL
+            self.addressFamily = addressFamily
+        }
+    }
+
+    /// Reconnect options.
+    public struct ReconnectOptions: Sendable, Hashable {
+        /// The initial time to wait before reconnecting to a broker after the connection has been closed. The time is increased exponentially until reconnect.backoff.max.ms is reached. -25% to +50% jitter is applied to each reconnect backoff. A value of 0 disables the backoff and reconnects immediately.
+        /// Default: `100`
+        public var backoffMilliseconds: Int = 100
+
+        /// The maximum time to wait before reconnecting to a broker after the connection has been closed.
+        /// Default: `10000`
+        public var backoffMaxMilliseconds: Int = 10000
+
+        public init(
+            backoffMilliseconds: Int = 100,
+            backoffMaxMilliseconds: Int = 10000
+        ) {
+            self.backoffMilliseconds = backoffMilliseconds
+            self.backoffMaxMilliseconds = backoffMaxMilliseconds
+        }
+    }
+
+    /// SSL options.
+    public struct SSLOptions: Sendable, Hashable {
+        /// Path to client's private key (PEM) used for authentication.
+        public var keyLocation: String = ""
+
+        /// Private key passphrase (for use with ssl.key.location).
+        public var keyPassword: String = ""
+
+        /// Path to client's public key (PEM) used for authentication.
+        public var certificateLocation: String = ""
+
+        /// File or directory path to CA certificate(s) for verifying the broker's key. Defaults: On Windows the system's CA certificates are automatically looked up in the Windows Root certificate store. On macOS this configuration defaults to probe. It is recommended to install openssl using Homebrew, to provide CA certificates. On Linux install the distribution's ca-certificates package. If OpenSSL is statically linked or ssl.ca.location is set to probe a list of standard paths will be probed and the first one found will be used as the default CA certificate location path. If OpenSSL is dynamically linked the OpenSSL library's default path will be used (see OPENSSLDIR in openssl version -a).
+        public var CALocation: String = ""
+
+        /// Path to CRL for verifying broker's certificate validity.
+        public var CRLLocation: String = ""
+
+        /// Path to client's keystore (PKCS#12) used for authentication.
+        public var keystoreLocation: String = ""
+
+        /// Client's keystore (PKCS#12) password.
+        public var keystorePassword: String = ""
+    }
+
+    /// SASL options.
+    public struct SASLOptions: Sendable, Hashable {
+        /// SASL mechanism to use for authentication.
+        public var mechanism: KafkaConfiguration.SASLMechanism?
```

Comment:
We should nest this type into this one right?

Reply:
Typed security config implemented in #88